### PR TITLE
Add wallet sync documentation

### LINKED ISSUE #2486: Add WalletSync documentation similar to ChainSync
In #2461 i added `WalletSync`. 

We have a nice example of how to use `ChainSync` to sync your chainstate with an arbitrary data source, we should document the same with the wallet. 

https://bitcoin-s.org/docs/next/chain/filter-sync#syncing-block-filters-against-bitcoind

### PR DESCRIPTION
**Pull request by Christewart**
fixes #2486 

This is kind of blocked on #2564 as I need to update the example rpc call to include blockhash/block height information.

### CODE REVIEW

File: docs/wallet/wallet-sync.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+{
+  "wallet": {
+    "keymanager": {
+      "rootXpub": "..."
+    },
+    "xpub": "...",
+    "hdPath": "m/84'/0'/0'"
+  }
+}
+```
```
Comment (benthecarman) [inline docs/wallet/wallet-sync.md]: imo this doesn't really make sense to include here. If you are using bitcoin-s as a library then you won't have access to the cli/server

File: docs/wallet/wallet-sync.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+---
+
+## High level wallet state
+
+Our wallet infrastructure has a specific table called `state_descriptors`.
+This tracks chain state for our wallet.
+Here is an example of the contents of this table
+
+>sqlite> select * from state_descriptors;
+SyncHeight|0000000000000000000134aa9e949ea1d053042b8dfa59bdc73b0322a88f009e 665741
```
Comment (benthecarman) [inline docs/wallet/wallet-sync.md:13]: Instead of this, you can use `wallet.getSyncDescriptorOpt`

**Review by benthecarman — CHANGES_REQUESTED** [review]

File: docs/wallet/wallet-sync.md (hunk 1 above)
Comment (Christewart) [reply docs/wallet/wallet-sync.md]: I really liked this section on the rescan docs, i was trying to replicate it here 

https://bitcoin-s.org/docs/next/wallet/wallet-rescan#rescan-from-cli

maybe I should have an appropriate heading like we do there? 

>Wallet state from the cli

or something of that nature?

Comment (benthecarman) [reply docs/wallet/wallet-sync.md]: Yeah that would make more sense

**Review by benthecarman — APPROVED** [review]

**Pull request merged**